runtime/jailer_integ_test.go

```diff
@@ -0,0 +1,158 @@
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/stretchr/testify/require"
+
+	_ "github.com/firecracker-microvm/firecracker-containerd/firecracker-control"
+	"github.com/firecracker-microvm/firecracker-containerd/internal"
+	"github.com/firecracker-microvm/firecracker-containerd/proto"
+	fccontrol "github.com/firecracker-microvm/firecracker-containerd/proto/service/fccontrol/ttrpc"
+	"github.com/firecracker-microvm/firecracker-containerd/runtime/firecrackeroci"
+)
+
+func TestDisableJailing_Isolated(t *testing.T) {
+	internal.RequiresIsolation(t)
+	const vmID = 0
+
+	ctx := namespaces.WithNamespace(context.Background(), "default")
+
+	client, err := containerd.New(containerdSockPath, containerd.WithDefaultRuntime(firecrackerRuntime))
+	require.NoError(t, err, "unable to create client to containerd service at %s, is containerd running?", containerdSockPath)
+	defer client.Close()
+
+	image, err := client.Pull(ctx, guestDockerImage, containerd.WithPullUnpack, containerd.WithPullSnapshotter(naiveSnapshotterName))
+	require.NoError(t, err, "failed to pull image %s, is the the %s snapshotter running?", guestDockerImage, naiveSnapshotterName)
+
+	// TODO once Noah's immutable rootfs change is merged, we can use that as our rootfs for all VMs instead of copying
+	// one per-VM
+	rootfsBytes, err := ioutil.ReadFile(defaultVMRootfsPath)
+	require.NoError(t, err, "failed to read rootfs file")
+	rootfsPath := fmt.Sprintf("%s.%d", defaultVMRootfsPath, vmID)
+	err = ioutil.WriteFile(rootfsPath, rootfsBytes, 0600)
+	require.NoError(t, err, "failed to copy vm rootfs to %s", rootfsPath)
+	err = os.Chmod(rootfsPath, 0666)
+	require.NoError(t, err, "failed to file permissions %s", rootfsPath)
+
+	tapName := fmt.Sprintf("tap%d", vmID)
+	err = createTapDevice(ctx, tapName)
+	require.NoError(t, err, "failed to create tap device for vm %d", vmID)
+
+	containerName := fmt.Sprintf("%s-%d", t.Name(), time.Now().UnixNano())
+	snapshotName := fmt.Sprintf("%s-snapshot", containerName)
+
+	pluginClient, err := ttrpcutil.NewClient(containerdSockPath + ".ttrpc")
+	require.NoError(t, err, "failed to create ttrpc client")
+
+	fcClient := fccontrol.NewFirecrackerClient(pluginClient.Client())
+	_, err = fcClient.CreateVM(ctx, &proto.CreateVMRequest{
+		VMID: strconv.Itoa(vmID),
+		RootDrive: &proto.FirecrackerDrive{
+			PathOnHost:   rootfsPath,
+			IsReadOnly:   false,
+			IsRootDevice: true,
+		},
+		NetworkInterfaces: []*proto.FirecrackerNetworkInterface{
+			{
+				HostDevName: tapName,
+				MacAddress:  vmIDtoMacAddr(uint(vmID)),
+				AllowMMDS:   true,
+			},
+		},
+		ContainerCount: 1,
+		JailerConfig: &proto.JailerConfig{
+			State: JailingOff,
+		},
+	})
+	require.NoError(t, err, "failed to create VM")
+
+	testTimeout := 60 * time.Second
+	newContainer, err := client.NewContainer(ctx,
+		containerName,
+		containerd.WithSnapshotter(naiveSnapshotterName),
+		containerd.WithNewSnapshot(snapshotName, image),
+		containerd.WithNewSpec(
```

Comment:
A lot of the configuration for the container here is particular to the block device test and shouldn't be needed